image/png.ksy

```diff
@@ -292,6 +305,68 @@ types:
       - id: text_datastream
         process: zlib
         size-eos: true
+  animation_control_chunk:
+    doc-ref: https://wiki.mozilla.org/APNG_Specification#.60acTL.60:_The_Animation_Control_Chunk
+    seq:
+      - id: num_frames
+        type: u4
+        doc: Number of frames
+      - id: num_plays
+        type: u4
+        doc: Number of times to loop, 0 indicates infinite looping.
+  frame_control_chunk:
+    doc-ref: https://wiki.mozilla.org/APNG_Specification#.60fcTL.60:_The_Frame_Control_Chunk
+    seq:
+      - id: sequence_number
+        type: u4
+        doc: Sequence number of the animation chunk
+      - id: width
+        type: u4
+        valid:
+          max: _root.ihdr.width
+        doc: Width of the following frame
+      - id: height
+        type: u4
+        valid:
+          max: _root.ihdr.height
+        doc: Height of the following frame
+      - id: x_offset
+        type: u4
+        valid:
+          max: _root.ihdr.width - width
+        doc: X position at which to render the following frame
+      - id: y_offset
+        type: u4
+        valid:
+          max: _root.ihdr.height - height
+        doc: Y position at which to render the following frame
+      - id: delay_num
+        type: u2
+        doc: Frame delay fraction numerator
+      - id: delay_den
+        type: u2
+        doc: Frame delay fraction denominator
+      - id: dispose_op
+        type: u1
+        enum: dispose_op_values
+        doc: Type of frame area disposal to be done after rendering this frame
+      - id: blend_op
+        type: u1
+        enum: blend_op_values
+        doc: Type of frame area rendering for this frame
+    instances:
+      delay:
+        value: 'delay_num / (delay_den == 0 ? 100.0 : delay_den)'
+        doc: delay in seconds
+  frame_data_chunk:
+    doc-ref: https://wiki.mozilla.org/APNG_Specification#.60fdAT.60:_The_Frame_Data_Chunk
+    seq:
+      - id: sequence_number
+        type: u4
+        doc: Sequence number of the animation chunk
+      - id: frame_data
+        size-eos: true
+        doc: Frame data for the frame
```

Comment:
https://wiki.mozilla.org/APNG_Specification#.60fdAT.60:_The_Frame_Data_Chunk:

> At least one `fdAT` chunk is required for each frame. The compressed datastream is then the **concatenation** of the contents of the data fields **of all the `fdAT` chunks within a frame**.

This is currently not clear from the KSY spec, so please include it into this `doc`. It also applies to the `IDAT` chunk, but this is currently handled just implicitly as a raw byte array, so there is no space for `doc`:

https://github.com/kaitai-io/kaitai_struct_formats/blob/a4f0d393c97163708d53b557b4c46c23f7de9300/image/png.ksy#L64

Sample files using this data splitting across multiple `IDAT`/`fdAT` chunks are here: [https://philip.html5.org/tests/apng/tests.html#:~:text=IDAT, fdAT splitting](https://philip.html5.org/tests/apng/tests.html#:~:text=IDAT,%20fdAT%20splitting)